src/components/Composer/index.js

```diff
@@ -306,11 +306,18 @@ class Composer extends React.Component {
      * @param {ClipboardEvent} event
      */
     handlePaste(event) {
-        if (!this.props.checkComposerVisibility()) {
+        event.preventDefault();
+
+        const isVisible = this.props.checkComposerVisibility();
+        const isFocused = this.textInput.isFocused();
+
+        if (!(isVisible || isFocused)) {
```

Comment:
NAB This is understandable, but we can refactor this into a single line and add comment.